### PR TITLE
feat(shortcuts): make sidebar item icon configurable

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by djamaile** [comment]
This is super cool!

File: plugins/shortcuts/src/Shortcuts.tsx [hunk 1]
```diff
@@ -26,9 +26,17 @@ import {
   SidebarItem,
   SidebarScrollWrapper,
 } from '@backstage/core-components';
-import { useApi } from '@backstage/core-plugin-api';
+import { IconComponent, useApi } from '@backstage/core-plugin-api';
 
-export const Shortcuts = () => {
+/**
+ * ShortcutsProps
```
Comment (freben) [inline plugins/shortcuts/src/Shortcuts.tsx:32]:
```suggestion
 * Props for the {@link Shortcuts} component.
```

**Review by freben — COMMENTED** [review]
Alright, let's just get a proper doc comment on it.

File: plugins/shortcuts/src/Shortcuts.tsx (hunk 1 above)
Comment (kuangp) [reply plugins/shortcuts/src/Shortcuts.tsx:32]: done!

**Review by freben — APPROVED** [review]

**Pull request merged**